src/components/VetList.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  TablePagination
+} from '@material-ui/core';
+import { findAllVets } from '../service/PetClinicService';
+
+const VetList = () => {
+  const [vetList, setVetList] = useState([{ id: '', firstName: '', lastName: '', specialties: [] }]);
+
+  useEffect(() => {
+    findAllVets()
+      .then((data) => {
+        setVetList(data);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }, []);
+
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(8);
+
+  const handleChangePage = (newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<any>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  return (
+    <div>
+      <TableContainer component={Paper}>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell>Name</TableCell>
+              <TableCell>Specialties</TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {vetList.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage).map((vet) => (
+              <TableRow key={vet.id}>
+                <TableCell>
+                  {vet.firstName} {vet.lastName}
+                </TableCell>
+                <TableCell>
+                  {vet.specialties.length > 0 ? vet.specialties.map((specialty) => specialty.name).join(', ') : 'none'}
```

Comment:
Kind of complicated here; try to simplify ( you can make a separate function then call it here with args).